joomla/joomla-cms
# [5.1] Convert mod_banners to service provider

### Summary of Changes
 
Converts the banners site module to service provider

### Testing Instructions

- create site module `Banners`
- check site output

Note: system test exists
https://github.com/joomla/joomla-cms/blob/5.1-dev/tests/System/integration/site/modules/mod_banners/Default.cy.js

### Actual result BEFORE applying this Pull Request

module works
![image](https://github.com/joomla/joomla-cms/assets/66922325/17cef297-ace1-4082-a216-ade959d3fcc7)

### Expected result AFTER applying this Pull Request

module works
![image](https://github.com/joomla/joomla-cms/assets/66922325/bd4e2c83-524f-41a1-8970-951c53679b57)

### Link to documentations
Please select:
